src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/prison/PrisonMergeEventProcessor.kt

```diff
@@ -22,11 +22,11 @@ class PrisonMergeEventProcessor(
   fun processEvent(domainEvent: DomainEvent) {
     val prisonNumber = domainEvent.getPrisonNumber()
     prisonerSearchClient.getPrisoner(prisonNumber)?.let {
-      val from: PersonEntity? =
-        personRepository.findByPrisonNumber(domainEvent.additionalInformation?.sourcePrisonNumber!!)
-      val to: PersonEntity = personService.processPerson(
-        it.doNotReclusterOnUpdate(),
-      ) { personRepository.findByPrisonNumber(prisonNumber) }
+      val from: PersonEntity? = personRepository.findByPrisonNumber(domainEvent.additionalInformation?.sourcePrisonNumber!!)
+      val reconciledPerson = PrisonPersonReconciler.reconcile(it, from)
+      val to: PersonEntity = personService.processPerson(reconciledPerson.doNotReclusterOnUpdate()) {
+        personRepository.findByPrisonNumber(prisonNumber)
```

Comment:
this is where we are overwriting the person data on the TO record with the data on the FROM record.

We might want to rethink how we do these merges to be honest.
There is something to be said for simply merging the records and not bothering to call for updates.
Syscon Sync will be sending them anyway in the fairly near future so the call will disappear then. We could potentially consider taking it out now